### PR TITLE
DT-6297 use latest graphql queries everywhere

### PR DESCRIPTION
**Pull request by vesameskanen**
This PR removes all references to deprecated graphQL fields. Especially:
- Unified vehicleParking is now used instead of separate carPark and bikePark. Lots of duplicate code removed.
- New time types for itineraries adopted

Changes also include component refactoring and misc bug fixes. Lots of repeating code removed.

Itinerary page related components no longer use moment library and the component tree is now collected into a single 'itinerary' folder.

After this PR, linter gives one warning : UI is already using the new PlanConnection.

### CODE REVIEW

File: app/util/legUtils.js [hunk 1]
```diff
@@ -3,6 +3,14 @@ import get from 'lodash/get';
 import { BIKEAVL_UNKNOWN } from './vehicleRentalUtils';
 import { getRouteMode } from './modeUtils';
 
+/**
+ * Get time as  milliseconds since the Unix Epoch
+ */
+export function legTime(lt) {
+  const t = lt.estimated?.time || lt.scheduledTime;
+  return Date.parse(t);
+}
```
Comment (optionsome) [inline app/util/legUtils.js:12]: Quite often we use this for moment. I wonder if it makes any difference to create a method that directly parses the ISO 8601 time to moment.

File: app/component/AirportCollectLuggageLeg.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ function AirportCollectLuggageLeg(props, { config }) {
     <div className="row itinerary-row">
       <div className="small-2 columns itinerary-time-column">
         <div className="itinerary-time-column-time">
-          {moment(props.leg.endTime).format('HH:mm')}
+          {moment(props.leg.end).format('HH:mm')}
```
Comment (optionsome) [inline app/component/AirportCollectLuggageLeg.js]: I think you need to use the legTime method here to parse the actual time.

File: app/util/legUtils.js (hunk 1 above)
Comment (vesameskanen) [reply app/util/legUtils.js:12]: I added new util which returns an actual time string in the  form UI frequently uses.

File: app/component/AirportCollectLuggageLeg.js (hunk 2 above)
Comment (vesameskanen) [reply app/component/AirportCollectLuggageLeg.js]: Right - I grepped 'moment(props.leg', found many of those and fixed them all.

File: app/component/ParkOrStationHeader.js [hunk 3]
```diff
@@ -43,8 +46,8 @@ const ParkOrBikeStationHeader = ({ parkOrStation, breakpoint }, { config }) => {
     });
   }, []);
 
-  const { name, bikeParkId, stationId } = parkOrStation;
-  const parkHeaderId = bikeParkId ? 'bike-park' : 'car_park';
+  const { name, stationId } = parkOrStation;
+  const parkHeaderId = parkType === 'bike' ? 'bike-park' : 'car_park';
```
Comment (optionsome) [inline app/component/ParkOrStationHeader.js]: This is doesn't change this but `-` vs `_` difference is a bit odd.

File: app/component/itinerary/Itinerary.js [hunk 4]
```diff
@@ -687,35 +690,25 @@ const Itinerary = (
...
                 values={{
                   vehicle: vehicleNames[0],
-                  departureTime: firstDeparture ? (
-                    <LocalTime time={firstDeparture.startTime} />
-                  ) : (
-                    'ggh'
-                  ),
+                  departureTime: firstDeparture
+                    ? legTimeStr(firstDeparture.start)
+                    : 'ggh',
```
Comment (optionsome) [inline app/component/itinerary/Itinerary.js]: I wonder what this `ggh` is.

File: app/component/itinerary/Legs.js [hunk 5]
```diff
@@ -151,144 +152,42 @@ export default class ItineraryLegs extends React.Component {
...
-            focusAction={this.focus(leg.from)}
-            changeHash={this.props.changeHash}
-            tabIndex={this.props.tabIndex}
-          />,
-        );
+        legs.push(<SubwayLeg {...transitLegProps} />);
       } else if (leg.mode === 'AIRPLANE') {
         legs.push(
           <AirportCheckInLeg
+            index={j - 0.5}
```
Comment (optionsome) [inline app/component/itinerary/Legs.js:181]: What do these fractional indices mean?

File: app/component/itinerary/Legs.js [hunk 6]
```diff
@@ -151,144 +152,42 @@ export default class ItineraryLegs extends React.Component {
...
-        legs.push(
-          <ViaLeg
-            index={j}
-            leg={leg}
-            arrivalTime={startTime}
-            focusAction={this.focus(leg.from)}
-            focusToLeg={this.focusToLeg(leg)}
-          />,
-        );
+        legs.push(<ViaLeg {...legProps} arrival={startTime} />);
```
Comment (optionsome) [inline app/component/itinerary/Legs.js:155]: This is a bit confusing that the arrival=startTime but it was like this before as well.

File: app/component/itinerary/Legs.js [hunk 7]
```diff
@@ -312,40 +211,23 @@ export default class ItineraryLegs extends React.Component {
           }
           bicycleWalkLeg = {
             duration: 0,
-            startTime: 0,
-            endTime: 0,
+            start: leg.start,
+            end: leg.start,
```
Comment (optionsome) [inline app/component/itinerary/Legs.js:215]: Why are these the same?

File: app/component/itinerary/TransitLeg.js [hunk 8]
```diff
@@ -53,8 +57,7 @@ const filterNextLegs = leg => {
     return [];
   }
   return leg.nextLegs.filter(
-    nextLeg =>
-      moment(nextLeg.startTime).diff(moment(leg.startTime), 'hours') < 12,
+    nextLeg => legTime(nextLeg.start) - legTime(leg.start) < 12 * 3600 * 1000, // 12 hours
```
Comment (optionsome) [inline app/component/itinerary/TransitLeg.js:60]: Can we use the leg duration here or are these legs not the legs returned from OTP (or are there additional legs generated by ui)?

**Review by optionsome — CHANGES_REQUESTED** [review]
We need to decide what to do with backward compatibility of parking pages as this changes which parking ids are accepted (previously non-feedscoped, now feedscoped).

I think there might be an issue with showing occupancy information at least on itinerary page (the icons don't show up when they do in dev).

Vehicle rental station availability (number of bicycles) is not shown on map.

On parking page, the selected parking is not highlighted on map.

On bicycle parking page, some capacity/availability number is displayed in the content as 0, it wasn't shown at all previously.

Stops near you buttons on frontpage don't show an alert which is visible in the view.

Vehicles are not shown on map in itinerary views.

Duration of a leg is shown as `NaN min` min with interlining trips (can be tested with for example kuopio config with /reitti/Vaahteratie%2024%2C%20Kuopio%3A%3A62.93671883702396%2C27.654132843017578/Korpitie%203%2C%20Kuopio%3A%3A62.85357911448784%2C27.66220092773438/?setTime=true&time=1713886213

File: app/component/itinerary/Legs.js (hunk 7 above)
Comment (vesameskanen) [reply app/component/itinerary/Legs.js:215]: Just two valid values. Prop must be given, used to be 0 for both.

File: app/component/itinerary/TransitLeg.js (hunk 8 above)
Comment (vesameskanen) [reply app/component/itinerary/TransitLeg.js:60]: There is no precomputed duraton as difference is computed from two different legs.

File: app/component/itinerary/Legs.js (hunk 6 above)
Comment (vesameskanen) [reply app/component/itinerary/Legs.js:155]: Yes, I guess that is somehow correct.

File: app/component/itinerary/Legs.js (hunk 5 above)
Comment (vesameskanen) [reply app/component/itinerary/Legs.js:181]: Leg needs a number index prop (it is used as react key). Computation does that.

File: app/component/itinerary/Itinerary.js (hunk 4 above)
Comment (vesameskanen) [reply app/component/itinerary/Itinerary.js]: Me too. I removed it.

File: app/component/ParkOrStationHeader.js (hunk 3 above)
Comment (vesameskanen) [reply app/component/ParkOrStationHeader.js]: Right, translation ids are inconsistent. I fixed that one and removes some unused translations at the same time.

**Comment by vesameskanen** [comment]
Listed bugs and some more fixed.

**Comment by vesameskanen** [comment]
Added also logic for stopping slow searches. 

There is still a question if search iteration should be stopped if response contains an error message. For example, short walk search takes a long time because walking filters away all transit itineraries. Query is repeated 4 times, every time with poor results. Error code is WALKING_BETTER_THAN_TRANSIT.

**Comment by optionsome** [comment]
We still need to do something about the backwards compatibility of parking pages and the issue with alert not showing on stops near you buttons seems to persists (can be tested with matka config, although might also be that somehow I'm using wrong package version). Otherwise looks like the issues were fixed and found no new issues.

**Comment by vesameskanen** [comment]
I seem to get same alert triangles in stopsnearyou buttons in opas.matka.fi and this branch.
Backward compatibility of park links: it is not difficult but requires a bit ugly hsl specific configuration. I wonder if such links exist in reality, and we should take the effort.

**Review by optionsome — APPROVED** [review]

**Pull request merged**